pkg/ccl/telemetryccl/telemetryccl_test: TestTelemetry failed

pkg/ccl/telemetryccl/telemetryccl_test.TestTelemetry [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16370881?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16370881?buildTab=artifacts#/) on master @ [89b1c1bfc5fb3edadd66eeb5e232ab7c07ee892d](https://github.com/cockroachdb/cockroach/commits/89b1c1bfc5fb3edadd66eeb5e232ab7c07ee892d):


```
        SELECT * FROM kv WHERE k = $1
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:24:
        exec [0 args]
        PREPARE p_join AS
        SELECT * FROM kv
        JOIN ab ON a = k
        WHERE v = $1
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:31:
        feature-list [0 args]
        sql.plan.type.force-custom
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:35:
        exec [0 args]
        SET plan_cache_mode = force_custom_plan
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:39:
        exec [0 args]
        SET index_recommendations_enabled = false
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:43:
        exec [0 args]
        SET application_name = 'testuser'
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:47:
        feature-usage [0 args]
        EXECUTE p_pk(100)
        ----
        sql.plan.type.force-custom
    datadriven.go:144: 
        testdata/telemetry/generic:52:
        feature-usage [0 args]
        EXECUTE p_join(1)
        ----
        sql.plan.type.force-custom
    datadriven.go:144: 
        testdata/telemetry/generic:58:
         SELECT * FROM kv WHERE v = 100
        expected:
        
        found:
        sql.plan.type.force-custom
        --- FAIL: TestTelemetry/generic/server (0.11s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128515 pkg/ccl/telemetryccl/telemetryccl_test: TestTelemetry failed [C-test-failure O-robot T-observability branch-release-24.2.0-rc release-blocker]
- #128171 pkg/ccl/telemetryccl/telemetryccl_test: TestTelemetry failed [C-test-failure O-robot P-1 T-observability branch-release-24.1 branch-release-24.2 release-blocker]
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41045

--- a/pkg/ccl/telemetryccl/testdata/telemetry/generic
+++ b/pkg/ccl/telemetryccl/testdata/telemetry/generic
@@ -54,15 +54,6 @@ EXECUTE p_join(1)
 ----
 sql.plan.type.force-custom
 
-# Non-prepared statements do not increment plan type counters.
-feature-usage
-SELECT * FROM kv WHERE v = 100
-----
-
-feature-usage
-SELECT * FROM kv WHERE v = 100
-----
-
 feature-list
 sql.plan.type.force-generic
 ----

